hls/decode-util_test.go

```diff
@@ -250,8 +250,8 @@ func TestReadMediaPlaylist(t *testing.T) {
 		if s.URI != newP.Segments[i].URI {
 			t.Errorf("Expected URI to be %s, but got %s", s.URI, newP.Segments[i].URI)
 		}
-		if s.Map != nil && !reflect.DeepEqual(s.Map, newP.Segments[i].Map) {
-			t.Errorf("Expected %d Segment Map tp be %v, but got %v", i, s.Map, newP.Segments[i].Map)
+		if !s.Map.Equal(newP.Segments[i].Map) {
```

Comment:
Joe Tsai covered this on his talk on gophercon!
https://about.sourcegraph.com/go/forward-compatible-go-code

I think it was the same day you opened this PR heh